pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -271,77 +136,50 @@ func Run() {
 	gc := time.NewTicker(downloadGCTime / 10)
 
 	for {
```

Comment:
Similar function can be called here.. Except - if we can abstract it to take in timers too??

Reply:
Yup. Same thing. When I wrote the above:

> Finish cleanup of the select on channels. There are too many repetitive statements there, which can be encapsulated into downloaderContext

This was _exactly_ the part I was referring to. :-)

Reply:
Still and all, I would prefer we simplify it in this large run, and then we can have a single PR focused on just the channel management, etc.

Reply:
Agree with Multiple PRs..